tools/ot/src/main/java/dk/alexandra/fresco/tools/ot/otextension/BristolRotBatch.java

```diff
@@ -0,0 +1,158 @@
+    int amountToPreprocess = computeExtensionSize(numMessages,
+        sender.getKbitLength(), sender.getLambdaSecurityParam());
+    Pair<List<StrictBitVector>, List<StrictBitVector>> messages = sender
+        .extend(amountToPreprocess);
+    List<StrictBitVector> rawZeroMessages = messages.getFirst();
+    List<StrictBitVector> rawOneMessages = messages.getSecond();
+    for (int i = 0; i < numMessages; i++) {
+      StrictBitVector zeroMessage = computeRandomMessage(rawZeroMessages.get(i),
+          sizeOfEachMessage);
+      StrictBitVector oneMessage = computeRandomMessage(rawOneMessages.get(i),
+          sizeOfEachMessage);
+      Pair<StrictBitVector, StrictBitVector> currentPair = new Pair<>(
+          zeroMessage, oneMessage);
+      res.add(currentPair);
+    }
+    return res;
+  }
+
+  @Override
+  public List<StrictBitVector> receive(StrictBitVector choiceBits,
+      int sizeOfEachMessage) {
+    // Initialize the underlying functionality if needed
+    if (!receiver.isInitialized()) {
+      receiver.initialize();
+    }
+    List<StrictBitVector> res = new ArrayList<>(choiceBits.getSize());
+    // Find how many OTs we need to preprocess
+    int amountToPreprocess = computeExtensionSize(choiceBits.getSize(),
+        receiver.getKbitLength(), receiver.getLambdaSecurityParam());
+    // Construct a new choice-bit vector of the original choices, padded with
+    // 0 choices if needed
+    byte[] extraByteChoices = Arrays.copyOf(choiceBits.toByteArray(),
+        amountToPreprocess / 8);
+    StrictBitVector extraChoices = new StrictBitVector(extraByteChoices,
+        amountToPreprocess);
+    List<StrictBitVector> messages = receiver.extend(extraChoices);
+    for (int i = 0; i < choiceBits.getSize(); i++) {
+      StrictBitVector newMessage = computeRandomMessage(messages.get(i),
+          sizeOfEachMessage);
+      res.add(newMessage);
+    }
+    return res;
+  }
+
+  /**
+   * Compute the minimal amount of OTs we have to preprocess in order to be sure
+   * to get "minSiz", <i>usable</i> OTs. This is not trivial since
+   * kbitLength+lambdaParam amount of OTs must be sacrificed in the underlying
+   * process.
+   *
+   * @param minSize
+   *          The amount of usable OTs we wish to have in the end
+   * @param kbitLength
+   *          The computational security parameter
+   * @param lambdaParam
+   *          The statistical security parameter
+   * @return The amount of OTs that must be extended such that the transposition
+   *         algorithm in correlated OT with errors work.
+   */
+  public static int computeExtensionSize(int minSize, int kbitLength,
+      int lambdaParam) {
+    // Increase the amount of OTs if needed, to ensure that the result is of the
+    // from 8*2^x for x > 1. I.e. s.t. that "minSize" >= 16
+    int newNum = Math.max(minSize, 16);
+    // Compute the number which will be passed on to the transposition algorithm
+    newNum = minSize + kbitLength + lambdaParam;
```

Comment:
Initialized value is overridden - what is the error, the overwritten value or the initialization

Reply:
I have fixed the issue in commit 5304bca0. Line 128 should have been minSize = Math.max(minSize, 16);